WebDevStudios/nextjs-wordpress-starter
# Feature/233 fe login rave

Closes #233

### Link

https://nextjs-wordpress-starter-m7cziyhzh-webdevstudios.vercel.app/

### Description

- Adds NextAuth to handle user auth/sessions
- Adds login, register, profile pages
    - Login redirects to profile if user is already logged in
    - Profile redirects to login if user is not logged in
    - "Sign out" link on profile logs user out and redirects to login
- Adds handling to hide/display login & profile links in menu depending on user session status
- Updates post comment handling to insert comment to WP db

### Screenshot

#### Login page
![Screenshot_2021-03-25 Next js WordPress Starter(1)](https://user-images.githubusercontent.com/36422618/112663097-0ccf3000-8e1e-11eb-975a-e6f16df8857e.png)

#### Registration page
![Screenshot_2021-03-25 Next js WordPress Starter](https://user-images.githubusercontent.com/36422618/112663098-0ccf3000-8e1e-11eb-86cf-7f6fac2bf5a6.png)

#### Profile page
![Screenshot_2021-03-25 Next js WordPress Starter(2)](https://user-images.githubusercontent.com/36422618/112663095-0ccf3000-8e1e-11eb-99a7-817304c728c0.png)

#### Comment form
![Screenshot_2021-03-26 Hello world - Next js WordPress Starter](https://user-images.githubusercontent.com/36422618/112663091-0c369980-8e1e-11eb-8d47-890c10f37649.png)

#### After comment submission
![Screenshot_2021-03-26 Hello world - Next js WordPress Starter(1)](https://user-images.githubusercontent.com/36422618/112663089-0c369980-8e1e-11eb-99cf-5d18542e2df4.png)

### Verification

https://nextjs-wordpress-starter-m7cziyhzh-webdevstudios.vercel.app/
- Click "Log In" link in menu (added to dev WP admin menu, will need to pull db for local testing)
- Enter WP user & pass or click "Create an Account" to register a new user (subscriber)
- On successful login/registration, site redirects to profile page
- Click "Sign out" to sign out of the site and be redirected to login page
- Go to any blog post (logged in or out) and submit comment form
- Depending on if user has previously approved comments, either confirmation message will display or comment will appear
- Go to WP admin to confirm comment was submitted
